source/core_http_client.c

```diff
@@ -553,6 +570,28 @@ static uint32_t getZeroTimestampMs( void )
 
 /*-----------------------------------------------------------*/
 
+static int8_t caseInsensitiveStringCmp( const char * str1,
+                                        const char * str2,
+                                        size_t n )
+{
+    size_t i = 0U;
+
+    assert( str1 != NULL );
+    assert( str2 != NULL );
+
+    for( i = 0U; i < n; i++ )
+    {
+        if( toupper( str1[ i ] ) != toupper( str2[ i ] ) )
```

Comment:
Are you certain about moving the `assert` outside the loop? An `assert` is easier to debug than a segfault...

Reply:
As we are making sure that n should be less than the length of either string, so there is no chance of segfault.

Reply:
That is what the documentation states but in practice, no one reads every detail, and someone makes a programming error eventually that would be easier to debug with an assert...